File: 387_first_unique_character_in_a_string/firstUniqChar.py
from collections import defaultdict


class Solution:
    def firstUniqChar(self, s: str) -> int:
        h_map = defaultdict(int)

        for i in s:
            h_map[i] += 1

        for i in range(len(s)):
            if h_map[s[i]] == 1:
                return i

        return -1

File: 387_first_unique_character_in_a_string/first_uniq_char.rs
use std::collections::HashMap;

impl Solution {
    pub fn first_uniq_char(s: String) -> i32 {
        let mut h_map = HashMap::new();

        for c in s.chars() {
            let e = h_map.entry(c).or_insert(0);
            *e += 1;
        }

        for (i, c) in s.chars().enumerate() {
            if *h_map.get(&c).unwrap() == 1 {
                return i as i32;
            }
        }

        -1
    }
}